lib/data/datasource/remote/api_service.dart

```diff
@@ -87,4 +90,39 @@ class ApiService {
       rethrow;
     }
   }
+
+  Future<DataState<List<MovieModel>>> getFavoriteMovies(
+      List<int> movieIds) async {
+    List<MovieModel> favoriteMovies = [];
+    try {
+      for (final id in movieIds) {
+        final response = await http.get(
+          Uri.parse(
+            '${MovieDetailsUiConstants.baseUrl}/$id',
+          ),
+          headers: {
+            HttpHeaders.authorizationHeader: authToken,
+          },
+        );
+        if (response.statusCode == HttpStatus.ok) {
+          favoriteMovies.add(
+            MovieModel.favoriteMovieFromJson(
+              json.decode(
+                response.body,
+              ),
+            ),
+          );
+        } else {
+          return const DataFailed(
+            favoriteMoviesErrorMsg,
+          );
+        }
+      }
+      return DataSuccess(
+        favoriteMovies,
+      );
+    } on Exception catch (e) {
+      rethrow;
```

Comment:
Why rethrowing instead of emitting a `DataFailed` state?

Reply:
i guess IDE suggested at first, but i changed it ! better this way